File: forge/src/main/java/org/valkyrienskies/create_interactive/forge/CreateInteractiveModForge.java
package org.valkyrienskies.create_interactive.forge;

import net.minecraft.client.renderer.ItemBlockRenderTypes;
import net.minecraft.client.renderer.RenderType;
import net.minecraft.server.level.ServerPlayer;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.event.entity.player.PlayerEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.DistExecutor;
import net.minecraftforge.fml.ModLoadingContext;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
import org.valkyrienskies.create_interactive.CreateInteractiveEventsClient;
import org.valkyrienskies.create_interactive.CreateInteractiveMod;
import org.valkyrienskies.create_interactive.GameContent;

@Mod(CreateInteractiveMod.MOD_ID)
public class CreateInteractiveModForge {

    public CreateInteractiveModForge() {
        // So we can @SubscribeEvent in this class
        MinecraftForge.EVENT_BUS.register(CreateInteractiveModForge.class);

        var MOD_BUS = FMLJavaModLoadingContext.get().getModEventBus();

        CreateInteractiveMod.INSTANCE.getREGISTRATE().registerEventListeners(MOD_BUS);

        CreateInteractiveMod.init();
        DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> CreateInteractiveMod::initClient);

        MOD_BUS.addListener(CreateInteractiveModForge::onClientSetup);


        ForgeConfigImpl.register(ModLoadingContext.get());
    }

    //Probably bad but oh well
    @SuppressWarnings("removal")
    private static void onClientSetup(FMLClientSetupEvent event) {
        ItemBlockRenderTypes.setRenderLayer(GameContent.INTERACT_ME.get(), RenderType.cutout());
        ItemBlockRenderTypes.setRenderLayer(GameContent.INTERACT_ME_NOT.get(), RenderType.cutout());
    }

    @SubscribeEvent
    public static void onPlayerJoin(PlayerEvent.PlayerLoggedInEvent joinEvent) {
        if (joinEvent.getEntity() instanceof ServerPlayer player) {
            CreateInteractiveEventsClient.INSTANCE.onPlayerJoin(player);
        }
    }
}

File: forge/src/main/kotlin/org/valkyrienskies/create_interactive/forge/CreateInteractiveModForge.kt
package org.valkyrienskies.create_interactive.forge

import net.minecraft.core.registries.Registries
import net.minecraft.world.item.CreativeModeTab
import net.minecraftforge.api.distmarker.Dist
import net.minecraftforge.client.event.EntityRenderersEvent.RegisterRenderers
import net.minecraftforge.eventbus.api.IEventBus
import net.minecraftforge.fml.DistExecutor
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext
import net.minecraftforge.registries.DeferredRegister
import org.valkyrienskies.create_interactive.CreateInteractiveMod
import org.valkyrienskies.create_interactive.CreateInteractiveMod.init
import org.valkyrienskies.create_interactive.CreateInteractiveMod.initClient
import org.valkyrienskies.mod.common.ValkyrienSkiesMod
import thedarkcolour.kotlinforforge.forge.MOD_BUS
import java.util.function.Supplier

@Mod(CreateInteractiveMod.MOD_ID)
class CreateInteractiveModForge {
    init {
        val MOD_BUS = FMLJavaModLoadingContext.get().modEventBus
        // val LOADING_CONTEXT = FMLJavaModLoadingContext.get()
        // Submit our event bus to let architectury register our content on the right time
        MOD_BUS.addListener { event: FMLClientSetupEvent? ->
            clientSetup(
                event
            )
        }
        MOD_BUS.addListener { event: RegisterRenderers ->
            entityRenderers(
                event
            )
        }

        CreateInteractiveModForgeHelper.registerRegistrate(MOD_BUS)

        DistExecutor.unsafeRunWhenOn(Dist.CLIENT) {
            Runnable {
                initClient()
            }
        }

        /*
        LOADING_CONTEXT.registerExtensionPoint(
            ConfigGuiFactory::class.java
        ) {
            ConfigGuiFactory { _: Minecraft?, parent: Screen? ->
                VS2KotlinHelper.createConfigScreenFor(
                    parent!!,
                    getRegisteredConfig(CreateInteractiveConfig::class.java)
                )
            }
        }
         */
        init()

        val deferredRegister = DeferredRegister.create(Registries.CREATIVE_MODE_TAB, CreateInteractiveMod.MOD_ID)
        deferredRegister.register("general", Supplier { return@Supplier CreateInteractiveModForgeHelper.createCreativeTab() })
        deferredRegister.register(MOD_BUS)
    }

    private fun clientSetup(event: FMLClientSetupEvent?) {
    }

    private fun entityRenderers(event: RegisterRenderers) {}

    companion object {
        fun getModBus(): IEventBus = MOD_BUS
    }
}
